### ISSUE TITLE
Class lada.handler does not exist

### ISSUE THREAD

**Issue by hendryb** [spiritix/lada-cache#96]
**after installing, which looked like this:**
.....@LAPTOP-AO0OQVIB MINGW64 ..... (master)
$ composer require spiritix/lada-cache 3.0.4
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 22 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-php80 (v1.17.1): Downloading (100%)
  - Installing symfony/deprecation-contracts (v2.1.3): Downloading (100%)
  - Installing psr/cache (1.0.1): Downloading (100%)
  - Installing psr/http-factory (1.0.1): Loading from cache
  - Installing psr/http-client (1.0.1): Downloading (100%)
  - Installing spiritix/lada-cache (v3.0.4): Downloading (100%)
  - Installing psr/event-dispatcher (1.0.0): Loading from cache
  - Installing symfony/filesystem (v5.1.2): Downloading (100%)
  - Installing symfony/dependency-injection (v5.1.2): Downloading (100%)
  - Installing symfony/config (v5.1.2): Downloading (100%)
  - Installing symfony/var-exporter (v5.1.2): Downloading (100%)
  - Installing symfony/cache-contracts (v2.1.3): Downloading (100%)
  - Installing symfony/cache (v5.1.2): Downloading (100%)
  - Installing symfony/framework-bundle (v4.4.10): Downloading (100%)
  - Installing spomky-labs/base64url (v2.0.1): Downloading (100%)
  - Installing spomky-labs/aes-key-wrap (v5.0.0): Downloading (100%)
  - Installing fgrosse/phpasn1 (v2.1.1): Downloading (100%)
  - Installing web-token/jwt-framework (v2.1.8): Downloading (100%)
  - Installing simplito/bigint-wrapper-php (1.0.0): Downloading (100%)
  - Installing simplito/bn-php (1.1.2): Downloading (100%)
  - Installing simplito/elliptic-php (1.0.6): Downloading (100%)
  - Installing alexlisenkov/laravel-web-push (1.0.2): Downloading (100%)
spiritix/lada-cache suggests installing ext-phpiredis (Allows faster serialization and deserialization of the Redis protocol)
symfony/dependency-injection suggests installing symfony/yaml
symfony/dependency-injection suggests installing symfony/expression-language (For using expressions in service container configuration)
symfony/dependency-injection suggests installing symfony/proxy-manager-bridge (Generate service proxies to lazy load them)
symfony/config suggests installing symfony/yaml (To use the yaml reference dumper)
symfony/framework-bundle suggests installing ext-apcu (For best performance of the system caches)
symfony/framework-bundle suggests installing symfony/form (For using forms)
symfony/framework-bundle suggests installing symfony/serializer (For using the serializer service)
symfony/framework-bundle suggests installing symfony/validator (For using validation)
symfony/framework-bundle suggests installing symfony/yaml (For using the debug:config and lint:yaml commands)
symfony/framework-bundle suggests installing symfony/property-info (For using the property_info service)
symfony/framework-bundle suggests installing symfony/web-link (For using web links, features such as preloading, prefetching or prerendering)
fgrosse/phpasn1 suggests installing php-curl (For loading OID information from the web if they have not bee defined statically)
web-token/jwt-framework suggests installing ext-sodium (Sodium is required for OKP key creation, EdDSA signature algorithm and ECDH-ES key encryption with OKP keys)
web-token/jwt-framework suggests installing bjeavons/zxcvbn-php (Adds key quality check for oct keys.)
web-token/jwt-framework suggests installing php-http/httplug (To enable JKU/X5U support.)
web-token/jwt-framework suggests installing php-http/httplug-bundle (To enable JKU/X5U support.)
web-token/jwt-framework suggests installing php-http/message-factory (To enable JKU/X5U support.)
web-token/jwt-framework suggests installing symfony/serializer (Use the Symfony serializer to serialize/unserialize JWS and JWE tokens.)
Package jakub-onderka/php-console-color is abandoned, you should avoid using it. Use php-parallel-lint/php-console-color instead.
Package jakub-onderka/php-console-highlighter is abandoned, you should avoid using it. Use php-parallel-lint/php-console-highlighter instead.
Package moontoast/math is abandoned, you should avoid using it. Use brick/math instead.
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: alexlisenkov/laravel-web-push
Discovered Package: barryvdh/laravel-debugbar
Discovered Package: beyondcode/laravel-dump-server
Discovered Package: beyondcode/laravel-er-diagram-generator
Discovered Package: biscolab/laravel-recaptcha
Discovered Package: elibyy/tcpdf-laravel
Discovered Package: fideloper/proxy
Discovered Package: intervention/validation
Discovered Package: kkomelin/laravel-translatable-string-exporter
Discovered Package: laravel/telescope
Discovered Package: laravel/tinker
Discovered Package: mcamara/laravel-localization
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Discovered Package: shivella/laravel-bitly
Discovered Package: spatie/geocoder
Discovered Package: spatie/laravel-permission
Discovered Package: themsaid/laravel-mail-preview
Package manifest generated successfully.

......@LAPTOP-AO0OQVIB MINGW64 .......... (master)
$ composer require spiritix/lada-cache 3.0.4
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Package jakub-onderka/php-console-color is abandoned, you should avoid using it. Use php-parallel-lint/php-console-color instead.
Package jakub-onderka/php-console-highlighter is abandoned, you should avoid using it. Use php-parallel-lint/php-console-highlighter instead.
Package moontoast/math is abandoned, you should avoid using it. Use brick/math instead.
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: alexlisenkov/laravel-web-push
Discovered Package: barryvdh/laravel-debugbar
Discovered Package: beyondcode/laravel-dump-server
Discovered Package: beyondcode/laravel-er-diagram-generator
Discovered Package: biscolab/laravel-recaptcha
Discovered Package: elibyy/tcpdf-laravel
Discovered Package: fideloper/proxy
Discovered Package: intervention/validation
Discovered Package: kkomelin/laravel-translatable-string-exporter
Discovered Package: laravel/telescope
Discovered Package: laravel/tinker
Discovered Package: mcamara/laravel-localization
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Discovered Package: shivella/laravel-bitly
Discovered Package: spatie/geocoder
Discovered Package: spatie/laravel-permission
Discovered Package: themsaid/laravel-mail-preview
Package manifest generated successfully.

......@LAPTOP-AO0OQVIB MINGW64 ........... (master)
$ php artisan vendor:publish

 Which provider or tag's files would you like to publish?:
  [0 ] Publish files from all providers and tags listed below
  [1 ] Provider: AlexLisenkov\LaravelWebPush\LaravelWebPushServiceProvider
  [2 ] Provider: Barryvdh\Debugbar\ServiceProvider
  [3 ] Provider: BeyondCode\DumpServer\DumpServerServiceProvider
  [4 ] Provider: BeyondCode\ErdGenerator\ErdGeneratorServiceProvider
  [5 ] Provider: Biscolab\ReCaptcha\ReCaptchaServiceProvider
  [6 ] Provider: Elibyy\TCPDF\ServiceProvider
  [7 ] Provider: Fideloper\Proxy\TrustedProxyServiceProvider
  [8 ] Provider: Illuminate\Foundation\Providers\FoundationServiceProvider
  [9 ] Provider: Illuminate\Notifications\NotificationServiceProvider
  [10] Provider: Illuminate\Pagination\PaginationServiceProvider
  [11] Provider: KKomelin\TranslatableStringExporter\Providers\ExporterServiceProvider
  [12] Provider: Laravel\Telescope\TelescopeServiceProvider
  [13] Provider: Laravel\Tinker\TinkerServiceProvider
  [14] Provider: Mcamara\LaravelLocalization\LaravelLocalizationServiceProvider
  [15] Provider: Shivella\Bitly\BitlyServiceProvider
  [16] Provider: Spatie\Geocoder\GeocoderServiceProvider
  [17] Provider: Spatie\Permission\PermissionServiceProvider
  [18] Provider: Themsaid\MailPreview\MailPreviewServiceProvider
  [19] Provider: Themsaid\MailPreview\MailProvider
  [20] Tag: config
  [21] Tag: laravel-errors
  [22] Tag: laravel-mail
  [23] Tag: laravel-notifications
  [24] Tag: laravel-pagination
  [25] Tag: migrations
  [26] Tag: telescope-assets
  [27] Tag: telescope-config
  [28] Tag: telescope-migrations
  [29] Tag: telescope-provider

**i got this error when opening a view using model with "use \Spiritix\LadaCache\Database\LadaCacheTrait;"**
exception: "ReflectionException"
file: ".....\vendor\laravel\framework\src\Illuminate\Container\Container.php"
line: 788
message: "Class lada.handler does not exist"

**Comment by spiritix** [comment]
Hi @hendryb, which Laravel version are you using?

**Comment by hendryb** [comment]
@spiritix i am using Laravel v5.8.38 and everything else works fine, except this package.

**Comment by spiritix** [comment]
I am not sure if Package Auto Discovery was already available in 5.8. Please try to manually register the service provider in config/app.php:

```php
    'providers' => array(
        // ...
        Spiritix\LadaCache\LadaCacheServiceProvider::class,
    )
```

**Comment by hendryb** [comment]
@spiritix That fixed it! Many MANY thanks!
